woocommerce/woocommerce
# Display notice at the top single variations

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #40597.

**ACs**

- [ ] If dismissed, it disappears from all tabs at once.
- [ ] If dismissed, it doesn’t appear on any other variation in this product. It reappears on other products, though.
- [ ] Clicking the link takes the user to the main product. If there are unsaved changes, we show the browser confirmation dialog.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a WooCommerce store with this branch and enable the product editor using **WooCommerce > Settings > Advanced > Features**
2. Go to **Products > Add New**
3. Go to `/wp-admin/admin.php?page=wc-admin&path=/add-product&tab=variations` and add some attributes
4. Wait until variations get generated
5. Then publish the product and save its `productId` and one `variationId` you like
6. Then go to `/wp-admin/admin.php?page=wc-admin&path=/product/{productId}/variation/{variationId}&tab=general` (replace the `{productId}` and `{variationId}` with the values picked in point 5)
7. After a few seconds, the following message should be shown.

![Screenshot 2023-10-11 at 09 37 49](https://github.com/woocommerce/woocommerce/assets/1314156/05648ae6-aca7-4136-a8c2-eea47cc5cdf0)

8. Instead of `Product 5` you should see the parent product name.
9. Verify the message is being shown in every tab.
10. Press the `X` and verify the message is not visible anymore in any tab.
11. Go to another variation (same product). The message shouldn't appear.
12. Go to another variation (different product). The message should be visible again. 



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
